samples/javascript_nodejs/21.corebot-app-insights/dialogs/mainDialog.js

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ComponentDialog, DialogSet, DialogTurnStatus, TextPrompt, WaterfallDialog } = require('botbuilder-dialogs');
+const { LuisRecognizer } = require('botbuilder-ai');
+const { TimexProperty } = require('@microsoft/recognizers-text-data-types-timex-expression');
+const { MessageFactory, InputHints } = require('botbuilder');
+
+const MAIN_WATERFALL_DIALOG = 'mainWaterfallDialog';
+
+class MainDialog extends ComponentDialog {
+    /**
+     * @param {*} luisRecognizer
+     * @param {*} bookingDialog
+     */
+    constructor(luisRecognizer, bookingDialog) {
```

Comment:
The changes in the constructor are not required for the purposes of this sample app, which is narrowly defined to the scenario of adding telemetry to the coreBot sample app, if these are not needed please revert them.

```JavaScript
    constructor(luisRecognizer, bookingDialog) {
        super('MainDialog');

        if (!luisRecognizer) throw new Error('[MainDialog]: Missing parameter \'luisRecognizer\' is required');
        this.luisRecognizer = luisRecognizer;

        if (!bookingDialog) throw new Error('[MainDialog]: Missing parameter \'bookingDialog\' is required');

        // Define the main dialog and its related components.
        // This is a sample "book a flight" dialog.
        this.addDialog(new TextPrompt('TextPrompt'))
            .addDialog(bookingDialog)
            .addDialog(new WaterfallDialog(MAIN_WATERFALL_DIALOG, [
                this.introStep.bind(this),
                this.actStep.bind(this),
                this.finalStep.bind(this)
            ]));

        this.initialDialogId = MAIN_WATERFALL_DIALOG;
    }
'''